### 19. try..catch, 에러 핸들링

- 에러 발생하면 스크립트는 즉시 중단, 콘솔에 에러 출력
- try...catch 사용하면 중단 방지



#### try...catch

```javascript
try {
    // 실행할 코드
} catch (err) {
    // 에러 코드
}
```

1. try 안의 코드 실행
2. 에러가 없다면, try 끝까지 실행 후, catch는 건너뜀
3. 에러가 있다면,  try 실행 중단 후, catch로 건너뜀



##### 예시

```javascript
let json = "{bad json}";

try {
    let user = JSON.parse(json); // catch로 넘어감
    console.log(user.name); // 동작x
} catch(err) {
    console.log('err 발생')
    console.log(err.name);
    console.log(err.message);
}
```



#### throw 연산자

- 에러를 생성

  ````javascript
  let json = '{"age": 30}';
  
  try {
      let user = JSON.parse(json);
      
      if(!user.name) {
          throw new SyntaxError('불완전데이터: 이름없음');
      }
      
      console.log(user.name)
  } catch(err) {
      console.log("JSON Error:" + err.message);
  }
  ````

  